File: misc/currying_function_base_conversion/convertBase.js
const n = "0123456789abcdefghijklmnopqrstuvwxyz";


function toBase(base) {
    function innerFunc(num) {
        if (num < base) {
            return n[num]
        }
        let quotient = Math.floor(num / base);
        let remainder = num % base;
        return innerFunc(quotient) + n[remainder]
    }
    return innerFunc;
}

function toDec(base) {
    function innerFunc(num) {
        let result = 0;
        for (let i = 0; i < num.length; i++) {
            if (parseInt(n.indexOf(num[num.length - 1 - i])) >= base) {
                return `Invalid number: Base ${base} should not contain digit ${num[num.length - 1 - i]}`
            }
            result += parseInt(n.indexOf(num[num.length - 1 - i])) * base ** i;
        }
        return result;
    }
    return innerFunc;
}

File: misc/currying_function_base_conversion/convert_base.py
n = "0123456789abcdefghijklmnopqrstuvwxyz"

def to_base(base):
    def inner_function(num):
        if num < base:
            return n[num]
        return inner_function(num // base) + n[num % base]
    return inner_function

def to_dec(base):
    def inner_function(num):
        result = 0
        if int(n.index(num[len(num) - 1 - i])) >= base:
            return f"Invalid number: Base {base} should not contain digit {n.index(num[len(num) - 1 - i])}."
        for i in range(len(num)):
            result += int(n.index(num[len(num) - 1 - i])) * base ** i
        return result
    return inner_function

